File: AA_15-2/uri-1923.cpp
// URI - Rerisson and Barbecue - 1923

#include <bits/stdc++.h>

using namespace std;

int main() {
	int n, g;
	cin >> n >> g;

	map<string, vector<string> > relations;

	for (int i = 0; i < n; i++) {
		string p1, p2;
		cin >> p1 >> p2;

		if (relations.find(p1) == relations.end() ) {
			relations[p1] = {p2};
		} else {
			relations[p1].push_back(p2);
		}

		if (relations.find(p2) == relations.end() ) {
			relations[p2] = {p1};
		} else {
			relations[p2].push_back(p1);
		}

	}

	int level = 1;
	vector<string> frontier;
	frontier.push_back("Rerisson");
	vector<string> ans;

	while (level <= g) {
		vector<string> next;
		for (int i = 0; i < frontier.size(); i++) {
			string u = frontier.at(i);

			for (int j = 0; j < u.size(); j++) {
				if 
			}
		}
	}

	return 0;
}

File: AA_15-2/uri-1923.py
# URI 1923 - Rerisson and Barbecue

n, g = map(int, raw_input().split())

friends = {}

for i in range(n):
	p1, p2 = raw_input().split()

	if friends.get(p1) == None:
		friends[p1] = [p2]
	else:
		friends[p1].append(p2)

	if friends.get(p2) == None:
		friends[p2] = [p1]
	else:
		friends[p2].append(p1)

level = {"Rerisson" : 0}
ans = []
i = 1
frontier = ["Rerisson"]
invited = 0

while i <= g and frontier:
	next = []

	for u in frontier:
		for v in friends[u]:
			if level.get(v) == None:
				ans.append(v)
				invited += 1
				level[v] = i
				next.append(v)

	frontier = next
	i += 1

print invited

invited_people = sorted(ans)

for i in range(invited):
	print invited_people[i]